# Part -1 Python Basics (Variables)

1. Print Your Name with your Father name and Date of birth using suitable escape sequence charactor

In [1]:
print(" Name:\t MUHAMMAD KASHIF\n Father Name:\t ALI MUHAMMAD KHATRI\n Date of birth: 02/Oct/1998 ")

 Name:	 MUHAMMAD KASHIF
 Father Name:	 ALI MUHAMMAD KHATRI
 Date of birth: 02/Oct/1998 


2. Write your small bio using variables and print it using print function

In [2]:
bio = "I'm a .NET developer with expertise in C#, ASP.NET, and building scalable applications. I focus on delivering high-quality, efficient solutions while staying updated with the latest technologies."
print(bio)

I'm a .NET developer with expertise in C#, ASP.NET, and building scalable applications. I focus on delivering high-quality, efficient solutions while staying updated with the latest technologies.


3. Write a program in which use all the operators we can use in Python

In [3]:
# 1. Arithmetic Operators
a = 50
b = 3
print("Arithmetic Operators:")
print("Addition:", a + b)       
print("Subtraction:", a - b)    
print("Multiplication:", a * b) 
print("Division:", a / b)       
print("Floor Division:", a // b) 
print("Modulus:", a % b)        
print("Exponentiation:", a ** b) 

# 2. Comparison Operators
print("\nComparison Operators:")
print("Equal:", a == b)        
print("Not Equal:", a != b)    
print("Greater Than:", a > b)  
print("Less Than:", a < b)     
print("Greater or Equal:", a >= b)
print("Less or Equal:", a <= b)  

Arithmetic Operators:
Addition: 53
Subtraction: 47
Multiplication: 150
Division: 16.666666666666668
Floor Division: 16
Modulus: 2
Exponentiation: 125000

Comparison Operators:
Equal: False
Not Equal: True
Greater Than: True
Less Than: False
Greater or Equal: True
Less or Equal: False


4. Completes the following steps of small task:
    - Mention Marks of English , Islamiat and Maths out of 100 in 3 different variables
    - Mention Variable of Total Marks and assign 300 to it
    - Calculate Percentage

In [4]:
English = 50
Islamiat = 70
Maths = 80 
Total_Marks= 300
Percentage = ((English+Islamiat+Maths)/Total_Marks)*100
print(Percentage,"%")

66.66666666666666 %


# Part -2 Python Basics (Conditional Statements)

1) A company decided to give bonus of 5% to employee if his/her year of service is more than 5 years.
Ask user for their salary and year of service and print the net bonus amount.

In [5]:
employee_workYear = float(input("Add your work years"))
employee_Salary = int(input("Add your monthly salary"))

if employee_workYear > 5:
    bonus = employee_Salary * 0.05
    print("Bonus Amount: ", bonus)
else:
    print("Bonus Amount: ", 0)

Add your work years 6
Add your monthly salary 50000


Bonus Amount:  2500.0



2) Write a program to check whether a person is eligible for voting or not. (accept age from user) if age is greater than 17 eligible otherwise not eligible

In [6]:
age = int(input("Your age"))
if age >= 18:
    print("Eligible to vote")
else:
    print("not eligible to vote")

Your age 17


not eligible to vote



3) Write a program to check whether a number entered by user is even or odd.

In [7]:
number = int(input("Add a number"))

if number  % 2 == 0:
    print ("number is even")
else:
    print ("number is odd")

Add a number 5


number is odd


4) Write a program to check whether a number is divisible by 7 or not.
Show Answer

In [8]:
number = int(input("Add a number"))

if number  % 7 == 0:
    print ("this number can be divisible by 7 ")
else:
    print ("this number can't be divisible by 7")

Add a number 14


this number can be divisible by 7 


5) Write a program to display 
"Hello" if a number entered by user is a multiple of five , otherwise print "Bye".

In [9]:
number = int(input("Add a number"))

if number  % 5 == 0:
    print ("hello")
else:
    print ("Bye")

Add a number 500


hello


7) Write a program to display the last digit of a number.


In [10]:
number = int(input("Add a number"))
lastDigit =number  % 10
print("your last digit" , lastDigit)

Add a number 764


your last digit 4


9) Take values of length and breadth of a rectangle from user and print if it is square or rectangle.

In [11]:
Length = int(input("Enter Length"))
Breadth = int(input("Enter Breadth"))

if Length == Breadth:
    print("It is a Square")
else:
    print("It is a Rectangle")

Enter Length 5
Enter Breadth 5


It is a Square


10) Take two int values from user and print greatest among them.

In [12]:
first_value = int(input("Enter first value"))
second_value = int(input("Enter second value"))

if first_value > second_value:
    print(first_value, "is greater.")
elif second_value > first_value:
    print(second_value, "is greater.")
else:
    print("Both numbers are equal.")
    

Enter first value 5
Enter second value 5


Both numbers are equal.


11) A shop will give discount of 10% if the cost of purchased quantity is more than 1000.
Ask user for quantity
Suppose, one unit will cost 100.
Judge and print total cost for user.


In [14]:
 quantity = int(input("Enter quantity"))

price = 100
totalcost = quantity * price

if totalcost > 1000:
    discount = totalcost * 0.1
    totalcost -= discount
    print("Total cost after discount :", totalcost)
else:
    print("Total cost :", totalcost)


Enter quantity 50


Total cost after discount : 4500.0


12) A school has following rules for grading system:

a. Below 25 - F

b. 25 to 45 - E

c. 45 to 50 - D

d. 50 to 60 - C

e. 60 to 80 - B

f. Above 80 - A

Ask user to enter marks and print the corresponding grade.

In [15]:
marks = int(input("Enter your Marks: "))

if marks> 80 and marks <= 100:
    print("Grade A")
elif marks > 60 and marks <= 80:
    print("Grade B")
elif marks > 50 and marks <= 60:
    print("Grade C")
elif marks > 45 and marks <= 50:
    print("Grade D")
elif marks > 25 and marks <=45:
    print("Grade E")
elif marks > 0 and marks <=25:
    print('Grade F')
else: 
    print('Invalid Input')

Enter your Marks:  25


Grade F


14)A student will not be allowed to sit in exam if his/her attendence is less than 75%.

Take following input from user

- Number of classes held

- Number of classes attended.

And print

- percentage of class attended

- Is student is allowed to sit in exam or not.


In [16]:
NumberOfClass = int(input("Enter total number of classes held: "))
NumberOfClassAttend = int(input("Enter number of classes attended: "))


attendance = (NumberOfClassAttend / NumberOfClass) * 100
print("Attendance Percentage", attendance, "%")

if attendance <= 75:
    print("Allowed to sit in exam.")
else:
    print("Not allowed to sit in exam.")
    



Enter total number of classes held:  50
Enter number of classes attended:  25


Attendance Percentage 50.0 %
Allowed to sit in exam.


15) Modify the above question to allow student to sit if he/she has medical cause. Ask user if he/she has medical cause or not ( 'Y' or 'N' ) and print accordingly.

In [27]:
NumberOfClass = int(input("Enter total number of classes held: "))
NumberOfClassAttend = int(input("Enter number of classes attended: "))
isMedicalcause = input("Any medical cause (Y/N): ").strip().upper()

attendance = (NumberOfClassAttend / NumberOfClass) * 100
print("Attendance Percentage:", attendance, "%")

if attendance >= 75:
    print("Allowed to sit in exam.")
elif isMedicalcause == 'Y':
    print("Allowed to sit in exam as attendance is below 75% due to medical cause.")
else:
    print("Not allowed to sit in exam.")

Enter total number of classes held:  50
Enter number of classes attended:  25
Any medical cause (Y/N):  Y


Attendance Percentage: 50.0 %
Allowed to sit in exam as attendance is below 75% due to medical cause.


16) Write a program to check if a year is leap year or not.

If a year is divisible by 4 then it is leap year but if the year is century year like 2000, 1900, 2100 then it must be divisible by 400.

In [30]:
year = int(input("Enter a year: "))

if (year % 4 == 0 and year % 100 != 0) or (year % 400 == 0):
    print("leap year.")
else:
    print("not a leap year.")

Enter a year:  1800


not a leap year.


17) Ask user to enter age, gender ( M or F ), marital status ( Y or N ) and then using following rules print their place of service.

if employee is female, then she will work only in urban areas.

if employee is a male and age is in between 20 to 40 then he may work in anywhere

if employee is male and age is in between 40 t0 60 then he will work in urban areas only.

And any other input of age should print "ERROR"

In [32]:
age = int(input("Enter your age: "))
gender = input("What's your gender (M/F): ")
# marital_status = input("Enter your marital status (Y/N): ")

if gender == "F":
    print("Will work only in urban areas.")
elif (gender == "M") and (age >=  20 and age <40):
    print("Can work anywhere")
elif (gender == "M") and (age >= 40 and age <60):
    print("Can work in urban areas only.")
else:
    print("ERROR")

Enter your age:  20
What's your gender (M/F):  F


Will work only in urban areas.


6) Write a program to calculate the electricity bill (accept number of unit from user) according to the following criteria :
     Unit                                                     Price  
uptp 100 units                                             no charge
Next 200 units                                              Rs 5 per unit
After 200 units                                             Rs 10 per unit
(For example if input unit is 350 than total bill amount is Rs.3500
(For example if input unit is 97 than total bill amount is Rs.0
(For example if input unit is 150 than total bill amount is Rs.750

In [ ]:

units = int(input("Enter the number of units consumed: "))


if units <= 100:
    bill = 0
elif units <= 300:
    bill = (units - 100) * 5
else:
    bill = (200 * 5) + (units - 300) * 10


print(f"Total bill amount is Rs.{bill}")

13) Take input of age of 3 people by user and determine oldest and youngest among them.

In [ ]:
# Taking input of ages from the user
age1 = int(input("Enter the age of first person: "))
age2 = int(input("Enter the age of second person: "))
age3 = int(input("Enter the age of third person: "))


if age1 >= age2 and age1 >= age3:
    oldest = age1
elif age2 >= age1 and age2 >= age3:
    oldest = age2
else:
    oldest = age3

if age1 <= age2 and age1 <= age3:
    youngest = age1
elif age2 <= age1 and age2 <= age3:
    youngest = age2
else:
    youngest = age3

print(f"The oldest person is {oldest} years old.")
print(f"The youngest person is {youngest} years old.")